### ISSUE TITLE
AWS export to web (upload) broken

### ISSUE THREAD

**Issue by arthuro555** [4ian/GDevelop#956]
## Describe the bug
When you use the export to Web (Upload online) option, the link will give out a black page. When you open the developer tools, you see that every .js files couldn't be loaded because they were not found (404) so the main engine gdjs is never defined and the game is unable to load.

## To Reproduce
1. Go to create a project
2. Click on export it with web (upload online)
3. Open the link

* Please include a link to a game if possible!
* If applicable, add screenshots to help explain your problem.
http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1551771408702/

## Other details
* Include any OS/browser version/smartphone that you're using
* Add any other context about the problem here.
I'm using the latest version of chrome on the latest version of windows 10. I'm not the only person with that problem some other persons reported it on a gdevelop discord server.

**Comment by Lizard-13** [comment]
Duplicate of https://github.com/4ian/GDevelop/issues/460 that is still flagged as "closed".

**Comment by Bouh** [comment]
Issue is still present.

All files with an folder in index.html are not correctly loaded.
I've export examples and empty project with a empty scene. All export are black.

![image](https://user-images.githubusercontent.com/1670670/61321345-9e923780-a80b-11e9-93e6-b896cb0801ee.png)

```
GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/libs/jshashtable.js net::ERR_ABORTED 404 (Not Found)
(index):31 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/libs/hshg.js net::ERR_ABORTED 404 (Not Found)
gd.js:12 Uncaught ReferenceError: Hashtable is not defined
    at gd.js:12
(anonymous) @ gd.js:12
gd-splash-image.js:1 Uncaught ReferenceError: gdjs is not defined
    at gd-splash-image.js:1
(anonymous) @ gd-splash-image.js:1
(index):32 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/libs/rbush.js net::ERR_ABORTED 404 (Not Found)
(index):49 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/commontools.js net::ERR_ABORTED 404 (Not Found)
(index):51 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/inputtools.js net::ERR_ABORTED 404 (Not Found)
(index):53 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/cameratools.js net::ERR_ABORTED 404 (Not Found)
(index):50 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/runtimescenetools.js net::ERR_ABORTED 404 (Not Found)
(index):52 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/objecttools.js net::ERR_ABORTED 404 (Not Found)
(index):54 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/soundtools.js net::ERR_ABORTED 404 (Not Found)
(index):55 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/storagetools.js net::ERR_ABORTED 404 (Not Found)
(index):56 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/stringtools.js net::ERR_ABORTED 404 (Not Found)
(index):57 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/windowtools.js net::ERR_ABORTED 404 (Not Found)
(index):59 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/pixi.js net::ERR_ABORTED 404 (Not Found)
(index):58 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/networktools.js net::ERR_ABORTED 404 (Not Found)
(index):60 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/pixi-filters-tools.js net::ERR_ABORTED 404 (Not Found)
(index):62 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/runtimescene-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):63 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/layer-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):64 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/pixi-image-manager.js net::ERR_ABORTED 404 (Not Found)
(index):65 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/spriteruntimeobject-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):61 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/runtimegame-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):66 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/loadingscreen-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):67 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/howler-sound-manager/howler.min.js net::ERR_ABORTED 404 (Not Found)
(index):68 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/howler-sound-manager/howler-sound-manager.js net::ERR_ABORTED 404 (Not Found)
(index):69 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/fontfaceobserver-font-manager/fontfaceobserver.js net::ERR_ABORTED 404 (Not Found)
(index):70 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/fontfaceobserver-font-manager/fontfaceobserver-font-manager.js net::ERR_ABORTED 404 (Not Found)
(index):31 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/libs/hshg.js net::ERR_ABORTED 404 (Not Found)
(index):32 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/libs/rbush.js net::ERR_ABORTED 404 (Not Found)
inputmanager.js:19 Uncaught ReferenceError: gdjs is not defined
    at inputmanager.js:19
(anonymous) @ inputmanager.js:19
timemanager.js:14 Uncaught ReferenceError: gdjs is not defined
    at timemanager.js:14
(anonymous) @ timemanager.js:14
runtimeobject.js:22 Uncaught ReferenceError: gdjs is not defined
    at runtimeobject.js:22
(anonymous) @ runtimeobject.js:22
profiler.js:7 Uncaught ReferenceError: gdjs is not defined
    at profiler.js:7
(anonymous) @ profiler.js:7
runtimescene.js:14 Uncaught ReferenceError: gdjs is not defined
    at runtimescene.js:14
(anonymous) @ runtimescene.js:14
scenestack.js:9 Uncaught ReferenceError: gdjs is not defined
    at scenestack.js:9
(anonymous) @ scenestack.js:9
polygon.js:13 Uncaught ReferenceError: gdjs is not defined
    at polygon.js:13
(anonymous) @ polygon.js:13
force.js:16 Uncaught ReferenceError: gdjs is not defined
    at force.js:16
(anonymous) @ force.js:16
layer.js:17 Uncaught ReferenceError: gdjs is not defined
    at layer.js:17
(anonymous) @ layer.js:17
timer.js:14 Uncaught ReferenceError: gdjs is not defined
    at timer.js:14
(anonymous) @ timer.js:14
runtimegame.js:15 Uncaught ReferenceError: gdjs is not defined
    at runtimegame.js:15
(anonymous) @ runtimegame.js:15
variable.js:14 Uncaught ReferenceError: gdjs is not defined
    at variable.js:14
(anonymous) @ variable.js:14
variablescontainer.js:15 Uncaught ReferenceError: gdjs is not defined
    at variablescontainer.js:15
(anonymous) @ variablescontainer.js:15
oncetriggers.js:15 Uncaught ReferenceError: gdjs is not defined
    at oncetriggers.js:15
(anonymous) @ oncetriggers.js:15
runtimebehavior.js:16 Uncaught ReferenceError: gdjs is not defined
    at runtimebehavior.js:16
(anonymous) @ runtimebehavior.js:16
spriteruntimeobject.js:18 Uncaught ReferenceError: gdjs is not defined
    at spriteruntimeobject.js:18
(anonymous) @ spriteruntimeobject.js:18
(index):49 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/commontools.js net::ERR_ABORTED 404 (Not Found)
(index):50 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/runtimescenetools.js net::ERR_ABORTED 404 (Not Found)
(index):51 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/inputtools.js net::ERR_ABORTED 404 (Not Found)
(index):52 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/objecttools.js net::ERR_ABORTED 404 (Not Found)
(index):53 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/cameratools.js net::ERR_ABORTED 404 (Not Found)
(index):54 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/soundtools.js net::ERR_ABORTED 404 (Not Found)
(index):55 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/storagetools.js net::ERR_ABORTED 404 (Not Found)
(index):56 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/stringtools.js net::ERR_ABORTED 404 (Not Found)
(index):57 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/windowtools.js net::ERR_ABORTED 404 (Not Found)
(index):58 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/events-tools/networktools.js net::ERR_ABORTED 404 (Not Found)
(index):59 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/pixi.js net::ERR_ABORTED 404 (Not Found)
(index):60 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/pixi-filters-tools.js net::ERR_ABORTED 404 (Not Found)
(index):61 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/runtimegame-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):62 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/runtimescene-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):63 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/layer-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):64 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/pixi-image-manager.js net::ERR_ABORTED 404 (Not Found)
(index):65 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/spriteruntimeobject-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):66 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/pixi-renderers/loadingscreen-pixi-renderer.js net::ERR_ABORTED 404 (Not Found)
(index):67 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/howler-sound-manager/howler.min.js net::ERR_ABORTED 404 (Not Found)
(index):68 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/howler-sound-manager/howler-sound-manager.js net::ERR_ABORTED 404 (Not Found)
(index):69 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/fontfaceobserver-font-manager/fontfaceobserver.js net::ERR_ABORTED 404 (Not Found)
(index):70 GET http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/fontfaceobserver-font-manager/fontfaceobserver-font-manager.js net::ERR_ABORTED 404 (Not Found)
code0.js:1 Uncaught ReferenceError: gdjs is not defined
    at code0.js:1
(anonymous) @ code0.js:1
data.js:1 Uncaught ReferenceError: gdjs is not defined
    at data.js:1
(anonymous) @ data.js:1
(index):83 Uncaught ReferenceError: gdjs is not defined
    at (index):83
    at (index):99
(anonymous) @ (index):83
(anonymous) @ (index):99
```

**Comment by 4ian** [comment]
I've tested and it worked properly: http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/

I think it might be related to the internet connection quality at the time of the upload. I should add some better retry mechanisms and error reporting.

**Comment by Bouh** [comment]
> I've tested and it worked properly: http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1563302124090/

For me it doesn't work.
(My connection internet is 646.5Mbit/s, with fiber)

**Comment by brendansarpong** [comment]
here's the link for mine: http://gd-games.s3-website-eu-west-1.amazonaws.com/game-1564671722923/

![upload error](https://user-images.githubusercontent.com/43975753/62305298-a974db00-b44d-11e9-9868-07e2dbc2d930.PNG)

^and there's the picture. 

![error snip](https://user-images.githubusercontent.com/43975753/62305525-230cc900-b44e-11e9-99b9-272c089e6b01.PNG)
^that is the console.

**Comment by 4ian** [comment]
Would you be able to reproduce it and send me the content of the GDevelop console (open the developer console from the View menu > Developer Console)?
If you're on Linux, the content of the terminal would also be useful.

**Comment by Bouh** [comment]
It looks like a missing file at a specific time

**Comment by brendansarpong** [comment]
What sub folder could I be missing? All the items for the game are in the same folder, and I exported it to a link. It's working in previews as well

**Comment by 4ian** [comment]
I'm more thinking of upload errors to AWS, but hard to tell

**Comment by Bouh** [comment]
I am thinking of the AWS service which does not provide the files correctly.

**Comment by brendansarpong** [comment]
How should I fix this

**Comment by 4ian** [comment]
> I am thinking of the AWS service which does not provide the files correctly.

No I don't think so, AWS is very reliable on this. If files are not served, it's that they have not been uploaded correctly.

I don't have a solution for now, you can work around by exporting to HTML5 and hosting it by yourself if it's really needed.

**Comment by Bouh** [comment]
we can close this issue, fixed in  #1179
